src/ControllerStatusIO.c

```diff
@@ -114,23 +171,40 @@ BOOL Ros_Controller_Initialize()
     }
 
     //get the robot calibration data for multi-robot systems
-    for (i = 0; i < MAX_ROBOT_CALIBRATION_FILES; i += 1)
+    const BOOL bCalibLoadedOk = Ros_Controller_LoadGroupCalibrationData(&g_Ros_Controller);
+    const size_t numBaseGroups = Ros_Controller_CountBaseGroups(&g_Ros_Controller);
+
+    //It's OK if calibration did not load (for single-group systems fi, or systems
+    //that haven't been calibrated), but post an alarm just in case IFF:
+    // - this is a multi-group system
+    // - there are groups other than base groups (so R1+R2, or R1+S1)
+    // - TF broadcast is enabled
+    // - none of the calibration files loaded, and
+    // - the warning has not been disabled
+    const BOOL bNeedToWarnAboutCalib =
+        g_Ros_Controller.numGroup > 1 &&
+        //NOTE: a base group 'takes care' of 1 regular group and itself, hence x2
+        (g_Ros_Controller.numGroup - 2*numBaseGroups) > 0 &&
```

Comment:
What if both robots are on a track, `R1+R2+B1+B2`? This would evaluate to false.

Reply:
Could you perhaps provide a few more examples of possible group configurations which you believe should raise the alarm, and a few which shouldn't?

I don't have sufficient information right now to derive a correct condition.

Reply:
@ted-miller: could a Station be calibrated against another Station? Or only Robots against Robots + Stations?

I'm thinking of just counting the nr of edges in a fully connected graph (recognising this is an undirected graph, so `/2`) where all `R` and `S` groups count as vertices. But if `S` groups can't be calibrated against each other (which I sort-of expect), I should subtract those from the nr of relations to take into account.

Reply:
No, I don't think you can calibrate station-to-station.

I think the logic is simply, "If there is more than one group, ignoring bases". There has to be at least one robot. After that, it doesn't matter how many robots/stations there are.

Reply:
I have an updated implementation here: https://github.com/Yaskawa-Global/motoros2/compare/main...gavanderhoorn:motoros2:warn_about_missing_calib. But I haven't had a chance to test it yet.

I'll update this PR if/when I do.

Reply:
I have a feeling the new unit tests are causing the MotoROS2 binary to run out of stack. [This test](https://github.com/Yaskawa-Global/motoros2/compare/main...gavanderhoorn:motoros2:warn_about_missing_calib#diff-c6374f863bbf386ebe4c03efe704d83985947777c2af2952604d085b44fdd4ccR279) specifically tries to allocate a `Controller` and four `CtrlGroup`s. It doesn't like that.

Reply:
Default stack size is supposedly 40k. Might be worth checking the `sizeof` some of those.

Reply:
According to `mpTaskCtrl.h`, we're currently using "the optimum value" for the stack. The next option would be `MP_STACK_LARGE_SIZE`, which is 50KB. But that just crashed my controller, so not sure what's going on.

> Might be worth checking the `sizeof` some of those.

that was indeed going to be my next step. I seem to remember especially `Controller` can be large, but I'm just mocking some simple `struct`s. Not even populating all fields/lists.

Reply:
@ted-miller: could you please test https://github.com/Yaskawa-Global/motoros2/compare/main...gavanderhoorn:motoros2:warn_about_missing_calib instead of this PR? Note: the branch has the same name, but it's on my fork and is a different version of the changes in this PR.

Ignore the tests (so make sure to *not* set `MOTOROS2_TESTING_ENABLE`), I still haven't had time to look into the crashes.

Reply:
My R1R2S1 configuration is broken right now. I'll need to re-init sometime tomorrow.

Reply:
(Unless the `WRONG DATA` alarms are somehow caused by the code. I'm assuming your crashes are not WRONG DATA alarms.)

Reply:
Unless you've still got `MOTOROS2_TESTING_ENABLE` defined, there should not be any errors/alarms raised by this, other than the `8013[16]` user alarm.